pkg/multitenant/cost_controller.go

```diff
@@ -96,6 +98,46 @@ func HasTenantCostControlExemption(ctx context.Context) bool {
 	return ctx.Value(exemptCtxValue) != nil
 }
 
+// TenantSideExternalIORecorder accounts for resources consumed when
+// writing or reading to external services such as an external storage
+// provider.
+type TenantSideExternalIORecorder interface {
+	// ExternalIOWriteWait waits for the RUs necessary to write
+	// the given number of bytes to an external service.
+	// necessary. Callers should call ExternaIOWriteSuccess or
+	// ExternalIOWriteFailure in the future to record the actual
+	// number of bytes written.
+	//
+	// If the context (or a parent context) was created using
+	// WithTenantCostControlExemption, the method is a no-op.
+	ExternalIOWriteWait(ctx context.Context, bytes int64)
+
+	// RecordExternalStroageEgressFailure returns RUs to the pool
+	// in the case of a Write that failed and wrote fewer bytes
+	// than expected. The RUs for the used number of bytes is
+	// recorded, the RUs for the unused bytes are returned to the
+	// pool.
+	//
+	// If the context (or a parent context) was created using
+	// WithTenantCostControlExemption, the method is a no-op.
+	ExternalIOWriteFailure(ctx context.Context, used int64, returned int64)
+
+	// RecordExternalStroageEgressSuccess records the usage of RUs associated with
+	// a write of the given size. This function does not wait.
```

Comment:
stray comment or missing method?

Reply:
These comments were all for typo'd versions of the old name. Should be fixed now.